src/sentry/incidents/endpoints/validators/numeric_comparison_validator.py

```diff
@@ -8,17 +8,9 @@
 
 
 class NumericComparisonConditionValidator(BaseDataConditionValidator):
-    comparison = serializers.FloatField(
-        required=True,
-        help_text="Comparison value to be compared against value from data.",
-    )
-    condition_result = serializers.ChoiceField(
-        choices=[
-            (DetectorPriorityLevel.HIGH, "High"),
-            (DetectorPriorityLevel.MEDIUM, "Medium"),
-            (DetectorPriorityLevel.LOW, "Low"),
-        ]
-    )
+    type = serializers.ChoiceField(choices=[(t.value, t.value) for t in Condition])
+    comparison = serializers.JSONField(required=True)
```

Comment:
what's different about having this as a `JSONField` and validating it in `validate_comparison` vs having it as a `FloatField`? i thought it would automatically raise a validation error if the type is not correct if it's `FloatField`

Reply:
The base class has these implemented as json (and the type is correctly json now). If this stayed a float there would be a mypy error.

Reply:
oh i see